evm_loader/program/src/executor/synced_state.rs

```diff
@@ -0,0 +1,275 @@
+use ethnum::{AsU256, U256};
+use maybe_async::maybe_async;
+use solana_program::instruction::Instruction;
+use solana_program::pubkey::Pubkey;
+use solana_program::rent::Rent;
+use std::collections::BTreeMap;
+
+use crate::account_storage::{AccountStorage, SyncedAccountStorage};
+use crate::error::{Error, Result};
+use crate::evm::database::Database;
+use crate::evm::Context;
+use crate::types::Address;
+
+use super::precompile_extension::PrecompiledContracts;
+use super::OwnedAccountInfo;
+
+pub struct SyncedExecutorState<'a, B: AccountStorage> {
+    pub backend: &'a mut B,
+    depth: usize,
+    transient_storage: BTreeMap<Address, BTreeMap<U256, [u8; 32]>>,
```

Comment:
Where do you handle reverts for `transient_storage`?

Reply:
Rewrote with the actions approach.